cachito/web/content_manifest.py

```diff
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import flask
+
+
+class ContentManifest:
+    """A content manifest associated with a Cacihto request."""
+
+    version = 1
+    json_schema_url = (
+        "https://raw.githubusercontent.com/containerbuildsystem/atomic-reactor/"
+        "f4abcfdaf8247a6b074f94fa84f3846f82d781c6/atomic_reactor/schemas/content_manifest.json"
+    )
+    unknown_layer_index = -1
+
+    def __init__(self, request=None):
+        """
+        Initialize ContentManifest.
+
+        :param Request request: the request to generate a ContentManifest for
+        """
+        self.request = request
+        self._packages = None
+        self._gopkg_data = {}
+        self._gomod_data = {}
+
+    def process_go_package(self, package, dependency):
+        """
+        Process go-package package.
+
+        :param Package package: the go-package package to process
+        :param Dependency dependency: the go-package package dependency to process
+        """
+        package = package.to_purl()
+        self._gopkg_data.setdefault(package, {"purl": package, "dependencies": [], "sources": []})
+        if dependency.type == "go-package":
+            icm_dependency = {"purl": dependency.to_purl()}
+            self._gopkg_data[package]["dependencies"].append(icm_dependency)
+
+    def process_gomod(self, package, dependency):
+        """
+        Process gomod package.
+
+        :param Package package: the gomod package to process
+        :param Dependency dependency: the gomod package dependency to process
+        """
+        self._gomod_data.setdefault(package.id, {"sources": [], "pkg": None})
+        if dependency.type == "gomod":
+            icm_source = {"purl": dependency.to_purl()}
+            self._gomod_data[package.id]["sources"].append(icm_source)
+        elif dependency.type == "go-package":
+            if self._gomod_data[package.id]["pkg"]:
+                raise RuntimeError(f'Multiple go-package dependencies for package "{package.id}"')
+            icm_package = dependency.to_purl()
+            self._gomod_data[package.id]["pkg"] = icm_package
+
+    def to_json(self):
```

Comment:
Could you please set `self._gopkg_data` and `self._gomod_data` to empty dictionaries at the start of this method? This will allow the `to_json` method to be run multiple times and get the same results.

This is what currently happens:
```python
>>> manifest = Request.query.get(1).content_manifest
>>> pprint(manifest.to_json())
{'image_contents': [{'dependencies': [{'purl': 'pkg:golang/github.com%2Fpkg%2Ferrors@v0.8.0'},
                                      {'purl': 'pkg:golang/github.com%2Fop%2Fgo-logging@v0.0.0-20160315200505-970db520ece7'},
                                      {'purl': 'pkg:golang/github.com%2FMasterminds%2Fsemver@v1.4.2'},
                                      {'purl': 'pkg:golang/gopkg.in%2Fyaml.v2@v2.2.2'},
                                      {'purl': 'pkg:golang/github.com%2Frelease-engineering%2Fretrodep%2Fv2%2Fretrodep%2Fglide@v2.0.2-0.20190619083402-c5360d4a2f5f'},
                                      {'purl': 'pkg:golang/golang.org%2Fx%2Ftools%2Fgo%2Fvcs@v0.0.0-20190325161752-5a8dccf5b48a'},
                                      {'purl': 'pkg:golang/github.com%2Frelease-engineering%2Fretrodep%2Fv2%2Fretrodep@v2.0.2-0.20190619083402-c5360d4a2f5f'}],
                     'purl': 'pkg:golang/github.com%2Frelease-engineering%2Fretrodep%2Fv2@v2.0.2-0.20190619083402-c5360d4a2f5f',
                     'sources': [{'purl': 'pkg:golang/github.com%2Fpkg%2Ferrors@v0.8.0'},
                                 {'purl': 'pkg:golang/github.com%2FMasterminds%2Fsemver@v1.4.2'},
                                 {'purl': 'pkg:golang/github.com%2Fkr%2Fpretty@v0.1.0'},
                                 {'purl': 'pkg:golang/github.com%2Fkr%2Fpty@v1.1.1'},
                                 {'purl': 'pkg:golang/github.com%2Fkr%2Ftext@v0.1.0'},
                                 {'purl': 'pkg:golang/github.com%2Fop%2Fgo-logging@v0.0.0-20160315200505-970db520ece7'},
                                 {'purl': 'pkg:golang/golang.org%2Fx%2Fcrypto@v0.0.0-20190308221718-c2843e01d9a2'},
                                 {'purl': 'pkg:golang/golang.org%2Fx%2Fnet@v0.0.0-20190311183353-d8887717615a'},
                                 {'purl': 'pkg:golang/golang.org%2Fx%2Fsys@v0.0.0-20190215142949-d0b11bdaac8a'},
                                 {'purl': 'pkg:golang/golang.org%2Fx%2Ftext@v0.3.0'},
                                 {'purl': 'pkg:golang/golang.org%2Fx%2Ftools@v0.0.0-20190325161752-5a8dccf5b48a'},
                                 {'purl': 'pkg:golang/gopkg.in%2Fcheck.v1@v1.0.0-20180628173108-788fd7840127'},
                                 {'purl': 'pkg:golang/gopkg.in%2Fyaml.v2@v2.2.2'}]}],
 'metadata': {'icm_spec': 'https://raw.githubusercontent.com/containerbuildsystem/atomic-reactor/f4abcfdaf8247a6b074f94fa84f3846f82d781c6/atomic_reactor/schemas/content_manifest.json',
              'icm_version': 1,
              'image_layer_index': -1}}
>>> pprint(manifest.to_json())
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/src/cachito/web/content_manifest.py", line 65, in to_json
    self.process_gomod(req_dep.package, req_dep.dependency)
  File "/src/cachito/web/content_manifest.py", line 52, in process_gomod
    raise RuntimeError(f'Multiple go-package dependencies for package "{package.id}"')
RuntimeError: Multiple go-package dependencies for package "1"
```